docs/proposals/image-provenance.md

```diff
@@ -0,0 +1,320 @@
+whether the Backend should be allowed to change its mind.  There is a tradeoff between responsiveness
+to changes in policy, versus keeping existing services running.  The two models that make sense are:
+
+- never change a policy, unless some external process has ensured no active objects depend on the to-be-forbidden
+  images.
+- change a policy and assume that transition to new image happens faster than the existing pods decay.
+
+## Ubernetes
+
+If two clusters share an image policy backend, then they will have the same policies.
+
+The clusters can pass different tokens to the backend, and the backend can use this to distinguish
+between different clusters.
+
+## Image tags and IDs
+
+Image tags are like: `myrepo/myimage:v1`.
+
+Image IDs are like: `myrepo/myimage@sha256:beb6bd6a68f114c1dc2ea4b28db81bdf91de202a9014972bec5e4d9171d90ed`.
+You can see image IDs with `docker images --no-trunc`.
+
+The Backend needs to be able to resolve tags to IDs (by talking to the images repo).
+If the Backend resolves tags to IDs, there is some risk that the tag-to-ID mapping will be
+modified after approval by the Backend, but before Kubelet pulls the image.  We will not address this
+race condition at this time.
+
+We will wait and see how much demand there is for closing this hole. If the community demands a solution,
+we may suggest one of these:
+
+1.  Use a backend that refuses to accept images that are specified with tags, and require users to resolve to IDs
+    prior to creating a pod template.
+   - kubectl could be modified to automate this process.
+   - a CI/CD system or templatingcould be used that maps IDs to tags before Deployment modification/creation.
+1. Audit logs from kubelets to see image IDs were actually run, to see if any unapproved images slipped through.
+1. Monitor tag changes in image repository for suspicious activity, or restrict remapping of tags after initial application.
+
+If none of these works well, we could do the following:
+
+- Image Policy Admission Controller adds new field to Pod, e.g. `pod.spec.container[i].imageID` (or an annotation).
+  and kubelet will enforce that both the imageID and image match the image pulled.
+
+Since this adds complexity and interacts with imagePullPolicy, we avoid adding the above feature initially.
+
+### Caching
+
+There will be a cache of decisions in the admission controller. 
+TTL will be user-controllable, but default to 1 hour for allows and 30s for denies.
+Low TTL for deny allows user to correct a setting on the backend and see the fix
+rapidly.  It is assumed that denies are infrequent.
+Caching allows permits RC to scale up services even during short unavailability of the webhook backend.
+The ImageReviewSpec is used as the key to the cache.
+
+In the case of a cache miss and timeout talking to the backend, the default is to allow Pod creation.
+Keeping services running is more important than a hypotetical threat from an un-verified image.
```

Comment:
Thinking out loud here: what if you fell back to using any existing "allow" cache as a whitelist if no contact with the backend can be established for a (long) period of time? This allows things that are already deployed to continue running without failing completely open.